js/viz/gauges/bar_guage/resolve_label_overlapping.js

```diff
@@ -0,0 +1,107 @@
+            .push({
+                series: {
+                    isStackedSeries: ()=> false,
+                    isFullStackedSeries: ()=> false
+                },
+                getLabels: ()=> [{
+                    isVisible: ()=> true,
+                    getBoundingRect: () => {
+                        const box = bar._text.getBBox();
+
+                        if(bar._text._lastCoords) {
+                            return {
+                                x: box.x + bar._text._lastCoords.x,
+                                y: box.y + bar._text._lastCoords.y,
+                                width: box.width,
+                                height: box.height,
+                            };
+                        }
+
+                        return bar._text.getBBox();
+                    },
+                    shift: (x, y)=> {
+                        const box = bar._text.getBBox();
+
+                        bar._text._lastCoords = { x: x - box.x, y: y - box.y };
+                        bar._text.attr({ translateX: x - box.x, translateY: y - box.y });
+                    },
+                    draw: ()=> bar.hideLabel(),
+                    getData: ()=> { return { value: bar.getValue() }; },
+                    hideInsideLabel: ()=> false,
+                }]
+            }
+            );
+        return r;
+    }, { left: [], right: [] });
+}
+
+export const cleanLabelsForShift = (bars, minY)=> {
+    bars.forEach(bar => {
+        const box = bar._text.getBBox();
+        const lastCoords = bar._text._lastCoords ? bar._text._lastCoords : { x: 0, y: 0 };
+        if(minY > box.y + lastCoords.y) {
+            bar.hideLabel();
+            return;
+        }
+    });
+};
+
+const ARC_COORD_PREC = 5;
+
+const getStartCoordsArc = function(x, y, innerR, outerR, startAngleCos, startAngleSin, endAngleCos, endAngleSin, isCircle, longFlag) {
+    return { x: (x + outerR * startAngleCos).toFixed(ARC_COORD_PREC), y: (y - outerR * startAngleSin).toFixed(ARC_COORD_PREC) };
+};
+
+export const drawConnector = (bars) => {
+    bars.forEach((bar) => {
+        const x = bar._bar.attr('x');
+        const y = bar._bar.attr('y');
+        const innerRadius = bar._bar.attr('innerRadius');
+        const outerRadius = bar._bar.attr('outerRadius');
+        const startAngle = bar._bar.attr('startAngle');
+        const endAngle = bar._bar.attr('endAngle');
+        const coordStart = getStartCoordsArc.apply(null, normalizeArcParams(x, y, innerRadius, outerRadius, startAngle, endAngle));
+        const xStart = Number(coordStart.x);
+        const yStart = Number(coordStart.y);
+        const box = bar._text.getBBox();
+        const lastCoords = bar._text._lastCoords ? bar._text._lastCoords : { x: 0, y: 0 };
+        if(bar._angle > 90) {
+            bar._line.attr({ points: [
+                xStart + 1, yStart + 1, box.x + box.width + lastCoords.x, box.y + box.height / 2 + lastCoords.y
```

Comment:
Why 1 in this, what it is? Why not the half of the connector width?

Reply:
Fixed, shift is now half the thickness of the connector